model/src/main/java/uk/gov/hmcts/payment/api/service/AccountServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.payment.api.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.oauth2.client.OAuth2RestOperations;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import uk.gov.hmcts.payment.api.dto.AccountDto;
+
+@Service
+public class AccountServiceImpl implements AccountService<AccountDto, String> {
+
+    @Autowired
+    private OAuth2RestOperations restTemplate;
+
+    @Value("${liberata.api.account.url}")
+    private String baseUrl;
+
+    @Override
+    public AccountDto retrieve(String pbaCode) {
+        try {
+            return restTemplate
+                .getForObject(baseUrl + "/" + pbaCode, AccountDto.class);
+        } catch (HttpClientErrorException ex) {
+            System.out.print(ex);
```

Comment:
why are we suppressing client errors..? Maybe catch all errors and raise a custom exception?